src/main/java/hudson/matrix/MatrixProject.java

```diff
@@ -727,7 +732,14 @@ public MatrixConfiguration getItem(Combination c) {
         return configurations.get(c);
     }
 
-    public File getRootDirFor(MatrixConfiguration child) {
+    /**
+     * Gets a root directory for the combination.
+     * Creates the whole directory hierarchy on-demand.
+     * @param combination
```

Comment:
I was quite confused by the implementation. Getter method with the automatic creation of directories seems to be quite crazy for me. It's being misused even within the plugin